ugc/ugc_tests/storage_tests.cpp

```diff
@@ -259,3 +264,43 @@ UNIT_TEST(StorageTest_LoadIndex)
   TEST(DeleteIndexFile(), ());
   TEST(DeleteUGCFile(), ());
 }
+
+UNIT_TEST(StorageTest_ContentTest)
+{
+  auto & builder = MwmBuilder::Builder();
+  m2::PointD const cafePoint(1.0, 1.0);
+  builder.Build({TestCafe(cafePoint)});
+  auto const cafeId = builder.FeatureIdForCafeAtPoint(cafePoint);
+  auto const oldUGC = MakeTestUGCUpdate(Time(chrono::hours(24 * 10)));
+  auto const newUGC = MakeTestUGCUpdate(Time(chrono::hours(24 * 300)));
+  Storage storage(builder.GetIndex());
+  storage.Load();
+  storage.SetUGCUpdate(cafeId, oldUGC);
+  storage.SetUGCUpdate(cafeId, newUGC);
+  TEST_EQUAL(storage.GetIndexesForTesting().size(), 2, ());
+  auto const toSendActual = storage.GetUGCToSend();
+
+  auto array = my::NewJSONArray();
+  vector<char> data;
+  {
+    using Sink = MemWriter<vector<char>>;
+    Sink sink(data);
+    SerializerJson<Sink> ser(sink);
+    ser(newUGC);
+  }
+
+  data.push_back('\0');
+  my::Json ugcNode(data.data());
+  auto embeddedNode = my::NewJSONObject();
+  ToJSONObject(*embeddedNode.get(), "data_version", cafeId.GetMwmVersion() );
+  ToJSONObject(*embeddedNode.get(), "mwm_name", cafeId.GetMwmName());
+  ToJSONObject(*embeddedNode.get(), "feature_id", cafeId.m_index);
```

Comment:
Wow, wow, can visitors help you to serialize this stuff?

Reply:
It can help in this particular case, but I don't use featureID as a source of these data outside of this unit test. There is a special struct which holds those data which is called ```Storage::UGCIndex```. And I use visitors for its serialization. But do you think it's a good idea to add a visitor into FeatureID only for one unit test?

Reply:
No, it's not a good idea, but can't you just override default UGC visitor and add an operator() for FeatureId?